Marlin/src/gcode/feature/trinamic/M911-M914.cpp

```diff
@@ -499,28 +499,27 @@
    * M914: Set StallGuard sensitivity.
    */
   void GcodeSuite::M914() {
-
     bool report = true;
     const uint8_t index = parser.byteval('I');
     LOOP_NUM_AXES(i) if (parser.seen(AXIS_CHAR(i))) {
       const int16_t value = parser.value_int();
       report = false;
       switch (i) {
-        #if X_SENSORLESS
+        #if X_SENSORLESS || X2_SENSORLESS
           case X_AXIS:
-            if (index < 2) stepperX.homing_threshold(value);
-            TERN_(X2_SENSORLESS, if (!(index & 1)) stepperX2.homing_threshold(value));
+            TERN_(X_SENSORLESS,  if (index < 2) stepperX.homing_threshold(value));
+            TERN_(X2_SENSORLESS, if (!index || index == 2) stepperX2.homing_threshold(value));
             break;
         #endif
-        #if Y_SENSORLESS
+        #if Y_SENSORLESS || Y2_SENSORLESS
           case Y_AXIS:
-            if (index < 2) stepperY.homing_threshold(value);
-            TERN_(Y2_SENSORLESS, if (!(index & 1)) stepperY2.homing_threshold(value));
+            TERN_(Y_SENSORLESS,  if (index < 2) stepperY.homing_threshold(value));
+            TERN_(Y2_SENSORLESS, if (!index || index == 2) stepperY2.homing_threshold(value));
             break;
         #endif
-        #if Z_SENSORLESS
+        #if Z_SENSORLESS || Z2_SENSORLESS || Z3_SENSORLESS || Z4_SENSORLESS
```

Comment:
This is ok, but it implies that you could have Z2 be sensorless when other (prior) Zs aren't sensorless. We don't want to allow that, so if it's currently being permitted then we ought to add a sanity-check to prevent it.